src/Features/Core/Portable/Diagnostics/Analyzers/NamingStyles/EditorConfigNamingStyleParser.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.CodeAnalysis.Diagnostics.Analyzers.NamingStyles
+{
+    internal static partial class EditorConfigNamingStyleParser
+    {
+        public static string GetNamingStylesStringFromDictionary(IReadOnlyDictionary<string, object> allRawConventions)
+        {
+            return ParseDictionary(allRawConventions).CreateXElement().ToString();
+        }
+
+        public static SerializableNamingStylePreferencesInfo ParseDictionary(IReadOnlyDictionary<string, object> allRawConventions)
```

Comment:
This is technically lossy - if there are symbol specs or naming styles that aren't referenced by a rule, we'll never create them.  Is that a problem? (I don't think so, as I guess we don't serialize these back out anywhere...)